runtime/domain.c

```diff
@@ -953,24 +975,61 @@ CAMLexport void caml_leave_blocking_section() {
     caml_plat_unlock(&self->interruptor.lock);
   }
 
-  caml_plat_lock(&self->domain_lock);
-  caml_leave_blocking_section_hook();
-  caml_process_pending_signals();
+  return;
 }
 
-CAMLexport void caml_enter_blocking_section() {
+CAMLexport void caml_bt_release_domain_lock(void)
+{
+  dom_internal* self = domain_self;
+  caml_plat_unlock(&self->domain_lock);
+  return;
+}
+
+CAMLexport void caml_bt_exit_ocaml(void)
+{
   dom_internal* self = domain_self;
 
   Assert(caml_domain_alone() || self->backup_thread_running);
 
-  caml_process_pending_signals();
-  caml_enter_blocking_section_hook();
   if (self->backup_thread_running) {
     atomic_store_rel(&self->backup_thread_msg, BT_IN_BLOCKING_SECTION);
     /* Wakeup backup thread if it is sleeping */
     caml_plat_signal(&self->domain_cond);
   }
-  caml_plat_unlock(&self->domain_lock);
+
+
+  return;
+}
+
+static void caml_enter_blocking_section_default(void)
+{
+  caml_bt_exit_ocaml();
+  caml_bt_release_domain_lock();
+  return;
+}
+
+static void caml_leave_blocking_section_default(void)
+{
+  caml_bt_enter_ocaml();
+  caml_bt_acquire_domain_lock();
```

Comment:
This function should be named `caml_acquire_domain_lock` shouldn't it? `bt` stands for backup thread, and we don't involve backup threads here, for example, in the case of exactly 1 domain in the program?